### ISSUE TITLE
Crate causes compilation failure when using the `sqlx` feature

### ISSUE THREAD

**Issue by mhutter** [shauncksm/base64id-rs#9]
## How to reproduce

```toml
# Cargo.toml
[package]
name = "ssss"
version = "0.1.0"
edition = "2021"

[dependencies]
base64id = { version = "0.3.1", features = ["sqlx"] }
```

Then try to build the crate:

```sh
cargo build -q
error: one of the features ['runtime-actix-native-tls', 'runtime-async-std-native-tls', 'runtime-tokio-native-tls', 'runtime-actix-rustls', 'runtime-async-std-rustls', 'runtime-tokio-rustls'] must be enabled
  --> /home/mh/.cargo/registry/src/index.crates.io-6f17d22bba15001f/sqlx-rt-0.6.3/src/lib.rs:11:1
   |
11 | / compile_error!(
12 | |     "one of the features ['runtime-actix-native-tls', 'runtime-async-...
13 | |      'runtime-tokio-native-tls', 'runtime-actix-rustls', 'runtime-asy...
14 | |      'runtime-tokio-rustls'] must be enabled"
15 | | );
   | |_^

error: could not compile `sqlx-rt` (lib) due to 1 previous error
```

**Comment by mhutter** [comment]
Note that the code above is the MCVE, but I actually first encountered the issue in a bigger project:

<details><summary>Full dependencies</summary>
<p>

```toml
[dependencies]
argon2 = { version = "0.5.3", features = ["std", "rand"] }
axum = { version = "0.7.7", default-features = false, features = ["http1", "tokio", "json", "macros"] }
base64id = { version = "0.3.1", features = ["rand", "serde", "sqlx"] }
serde = { version = "1.0.214", features = ["derive"] }
sqlx = { version = "0.8.2", default-features = false, features = ["runtime-tokio", "postgres", "time", "macros", "migrate"] }
time = { version = "0.3.36", features = ["serde"] }
tokio = { version = "1.41.0", default-features = false, features = ["rt-multi-thread", "net", "signal"] }
utoipa = { version = "5.2.0", features = ["axum_extras", "time"] }
```

</p>
</details>

I'll send a PR with a fix in a minute

**Comment by mhutter** [comment]
Another approach though would be to relax this crate's version requirements, for example like this:

```toml
[dependencies]
rand = { version = "0", optional = true, default-features = false}
serde = { version = "1", optional = true, default-features = false }
sqlx = { version = "0", optional = true, default-features = false, features = [ "macros" ] }
```

This would make Cargo select the latest version the user already has in their dependency tree to use for `base64id`. It of course does not handle incompatibilities introduced by `rand` or `sqlx` in future 0.x versions, however it seems that's the "more unlikely" scenario to happen.

**Comment by shauncksm** [comment]
Hi @mhutter

Thanks for logging an issue!

Would it be funny if I were to say that I've implemented the solution I had described above months ago in the derive branch. It's just unreleased as you've pointed out.

The reason for this of course... an _[epic round of procrastination](https://tenor.com/view/coding-humor-programmer-documentation-gif-18858223)_ on getting the docs up to date after the above changes. 

I've cracked open the project again, fixed up the docs to a degree I'm satisfied with (for the moment anyway).

I've also started work on adding support for unsigned types following your [suggestion](https://github.com/shauncksm/base64id-rs/issues/8#issuecomment-2458868952). I think I had considered doing this at Some Point(tm) in the future, though looking at it now I've found it's straight forward enough. 64 bit support is done. I hope to have 32 and 16 bit support there in the next day or so along with a 0.4 release shortly there after.

Perhaps all I needed was an issue to actually get #8 finished once and for all :sweat_smile:

**Comment by mhutter** [comment]
Let me know if I can help in any way!

(Then we can both procrastinate)

**Comment by shauncksm** [comment]
Will do! I think I'm good for the moment, finally got some time (since two days ago) to sit down and give it the time it needs.

**Comment by shauncksm** [comment]
Hey @mhutter! This should now be resolved in the newly released v0.4.1!

See the [migration guide](https://github.com/shauncksm/base64id-rs/blob/main/docs/MIGRATION.0.4.md) for how transition from the legacy concrete types to the new derive macro.

If I've added any [horrible catastrophes](https://github.com/shauncksm/base64id-rs/commit/79ce51d9b43c1c90e296573e3246da4fd302bdaf) please let me know.
